doc/manual/calling-c-and-fortran-code.rst

```diff
@@ -144,248 +149,551 @@ memory to be passed to the callee and filled in. Allocation of memory
 from Julia like this is generally accomplished by creating an
 uninitialized array and passing a pointer to its data to the C function.
 
-A prefix ``&`` is used to indicate that a pointer to a scalar argument
-should be passed instead of the scalar value itself (required for all
-Fortran function arguments, as noted above). The following
-example computes a dot product using a BLAS function.
+Creating C-Compatible Julia Function Pointers
+---------------------------------------------
 
-::
+It is possible to pass Julia functions to native functions that accept function
+pointer arguments by creating function pointers via the `cfunction` function.
 
-    function compute_dot(DX::Vector{Float64}, DY::Vector{Float64})
-      assert(length(DX) == length(DY))
-      n = length(DX)
-      incx = incy = 1
-      product = ccall( (:ddot_, "libLAPACK"),
-                      Float64,
-                      (Ptr{Int32}, Ptr{Float64}, Ptr{Int32}, Ptr{Float64}, Ptr{Int32}),
-                      &n, DX, &incx, DY, &incy)
-      return product
+Finally, you can use ``cfunction`` to actually generate a call to the
```

Comment:
noted. (this paragraph mirrored the section in ccall that I derived it from, but the ccall one actually had paragraphs in between)